File: contests/ICPC mx/repechaje/key_pad.cpp
#include<bits/stdc++.h>
using namespace std;
vector<char> keyPad = {'2', '2', '2', '3', '3', '3', '4', '4', '4', '5', '5', '5', '6', '6', '6', '7', '7', '7', '7', '8', '8', '8', '9', '9', '9', '9'};

int main(){
    int n,q; cin>>n>>q;
    string identifier, code;
    unordered_map<string, int> dict;

    for(int i=0; i<n; i++){
        string identifierStr = "";
        cin>>identifier;
        for(auto letter: identifier){
            identifierStr += keyPad[letter - 'a'];
        }
        dict[identifierStr] += 1;
    }

    for(int i=0; i<q; i++){
        cin>>code;
        cout<<dict[code]<<endl;
    }
    return 0;
}

File: contests/ICPC mx/repechaje/key_pad.py
from collections import defaultdict

keypad_codes = ["2", "2", "2", "3", "3", "3", "4", "4", "4", "5", "5", "5", "6", "6", "6", "7", "7", "7", "7", "8", "8", "8", "9", "9", "9", "9"]
""""
for i in range(2,10):
    repetitions = 4 if i in (7,9) else 3 
    aux = [i]*repetitions 
    keypad_codes.extend(aux)
"""
(n, q) = input().split(" ")
n = int(n)
q = int(q)
#print(keypad_codes)


dc = {}

for i in range(n):
    identifier = input()
    identifier_str = ""
    for letter in identifier:
        identifier_str += keypad_codes[ord(letter) - ord('a')]
    try:
        dc[identifier_str] += 1
    except KeyError:
        dc[identifier_str] = 1

codes = [] 
for i in range(q):
    code = input()
    try:
        print(dc[code])
    except KeyError:
        print(0)